analytics/analytics/job_processor/__init__.py

```diff
@@ -19,7 +20,12 @@
 from analytics.core.models import Job
 
 
+UNNECESSARY_JOB_REGEX = re.compile(r"No need to rebuild [^,]+, found hash match")
+
+
 def create_job(gl: gitlab.Gitlab, project: Project, gljob: ProjectJob) -> Job:
+    job_trace: str = gljob.trace().decode()
```

Comment:
Just curious if this is possibly a second or more retrieval of the job trace resulting from the web hook firing.  And if so, is that anything we should be concerned with?

Reply:
It is. I'm not super worried, but we can always consolidate jobs in the future if it becomes an issue.

FWIW, this is a rough distribution of our job log size (91% of logs are between 0-1MB):
```
0                   58,606     [91.0%] ██████████████████████████████████████████████████
1                   2,476      [ 3.8%] ██▏
2                   1,062      [ 1.6%] █
3                   819        [ 1.3%] ▊
4                   240        [ 0.4%] ▏
5                   73         [ 0.1%]
6                   66         [ 0.1%]
7                   41         [ 0.1%]
8                   78         [ 0.1%]
9                   125        [ 0.2%]
10                  823        [ 1.3%] ▊
```

Reply:
Ok, perfect, thanks for the details :+1: